apps/hpc-ftsadmin/src/app/pages/add-edit/flow-edit.tsx

```diff
@@ -0,0 +1,1857 @@
+      }
+    }
+  };
+
+  const createInputEntryExternal = (
+    src: string,
+    objType: string,
+    externalData: string
+  ) => {
+    const srcInputEntryEnum =
+      enumInputEntryVsSrcObjType[
+        src as keyof typeof enumInputEntryVsSrcObjType
+      ];
+    const key = srcInputEntryEnum[
+      objType as keyof typeof srcInputEntryEnum
+    ] as keyof InputEntriesType;
+
+    const inputEntry = {
+      category: forms.InputEntryCategoriesEnum.EXTERNAL,
+      value: externalData,
+      kind: forms.InputEntryKindsEnum.UNMATCHED,
+    };
+
+    const tempEntries = (inputEntries[key] as forms.InputEntryType[]).filter(
+      (entry) => entry.category !== forms.InputEntryCategoriesEnum.ACTIVE_FLOW
+    );
+    tempEntries.push(inputEntry);
+    (inputEntries[key] as forms.InputEntryType[]) = tempEntries;
+  };
+
+  useEffect(() => {
+    try {
+      const fetchData = async () => {
+        const parentResults = await Promise.all(
+          parentIds.map((id) =>
+            env.model.flows.getFlowREST({
+              id,
+            })
+          )
+        );
+
+        let isParkedParent = false;
+        const sources: Record<string, any[]> = {};
+
+        if (parentResults && parentResults[0] && parentResults[0].categories) {
+          const parent = parentResults[0];
+          isParkedParent = parent.categories.some(function (category) {
+            return (
+              category.group === 'flowType' &&
+              category.name.toLowerCase() === 'parked'
+            );
+          });
+
+          parent.flowObjects
+            .filter(function (flowObject) {
+              return flowObject.refDirection === 'source';
+            })
+            .forEach(function (flowObject) {
+              const objectType = flowObject.objectType;
+              const objTypeTail = objectType[flowObject.objectType.length - 1];
+
+              let objectTypePlural: string;
+              if (objTypeTail === 'y') {
+                objectTypePlural = objectType.replace('y', 'ies');
+              } else {
+                objectTypePlural = objectType + 's';
+              }
+              if (
+                !sources[objectTypePlural] ||
+                !sources[objectTypePlural].length
+              ) {
+                sources[objectTypePlural] = [];
+              }
+
+              const parentArray =
+                parent[objectTypePlural as keyof typeof parent];
+              if (parentArray && Array.isArray(parentArray)) {
+                const foundObject = parentArray.find((obj: any) => {
+                  return flowObject.objectID === obj.id;
+                });
+                if (foundObject) {
+                  sources[objectTypePlural].push(foundObject);
+                }
+              }
+            });
+
+          const srcUsageYear: usageYears.UsageYear[] = [];
+          const destUsageYear: usageYears.UsageYear[] = [];
+          parent.usageYears.map((item: usageYears.UsageYear) => {
```

Comment:
instead of using map, since it expects to return a value, and in any case you don't need it. Write it with a `for... of` loop:
`for(const item of parent.usageYears) ...`